server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java

```diff
@@ -921,22 +922,28 @@ private <T> void termsAggregator(
                         .size(size)
                         .shardSize(size)
                         .field("field")
+                        .showTermDocCountError(true)
                         .order(bucketOrder);
 
-                    Terms result = searchAndReduce(
-                        indexSearcher,
-                        new AggTestConfig(aggregationBuilder, fieldType).withSplitLeavesIntoSeperateAggregators(false)
-                    );
+                    AggTestConfig aggTestConfig = new AggTestConfig(aggregationBuilder, fieldType);
+                    Terms result = searchAndReduce(indexSearcher, aggTestConfig);
                     assertEquals(size, result.getBuckets().size());
+                    HashMap<Object, Long> expectedCounts = new HashMap<>();
                     for (int i = 0; i < size; i++) {
                         Map.Entry<T, Integer> expected = expectedBuckets.get(i);
+                        Object key = (valueType == ValueType.IP) ? String.valueOf(expected.getKey()).substring(1) : expected.getKey();
+                        expectedCounts.put(key, expected.getValue().longValue());
+                    }
+                    long docCountError = result.getDocCountError();
+                    for (int i = 0; i < size; i++) {
                         Terms.Bucket actual = result.getBuckets().get(i);
-                        if (valueType == ValueType.IP) {
-                            assertEquals(String.valueOf(expected.getKey()).substring(1), actual.getKey());
-                        } else {
-                            assertEquals(expected.getKey(), actual.getKey());
+                        long expectedCount = expectedCounts.getOrDefault(actual.getKey(), 0L);
+                        if (aggTestConfig.splitLeavesIntoSeparateAggregators() == false || indexReader.leaves().size() == 1) {
+                            assertThat("Count mismatch for " + actual.getKey(), actual.getDocCount(), equalTo(expectedCount));
+                        } else if (order == false) {
```

Comment:
I think it makes sense to force count ascending (aka `order == true`) into single segment mode.  Otherwise there's just no invariant we can test for it. I'm not 100% sure that testing count ascending in single segment mode proves anything, since prod rarely runs in single collector mode, but I also don't think we should allow sort by count ascending, so...

Reply:
When either `aggTestConfig.splitLeavesIntoSeparateAggregators() == false` or `indexReader.leaves().size() == 1` then we can assert docCounts when `order` is `false` or `true`?

Reply:
Yes, no matter the sort order, if there is only one sub-aggregator, there are no counts errors, so we can assert on every value. The older version of this test made this assumption, and also only used a single sub-aggregator. The new test allows for multiple aggregators, but can only make the assertions when the error is unbounded.

Reply:
So I've made a commit which does two things, adds comments clarifying all this, as well as implements the suggestion to force single-aggregator when order == true (ie. allow multiple aggregators for order == false).